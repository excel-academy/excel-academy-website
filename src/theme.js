import { themeColor } from '../site-config';

const purples = ['#7b79b0', '#6f6da5', '#eeeefc', '#1c2070'];
const greens = ['#d2eced', '#132c2d'];
const fontSizes = [12, 14, 16, 18, 20, 24, 30, 38, 62];
const space = [0, '2rem', '3rem', '4rem', '5rem', '7rem', '32rem', '40rem', '48rem', '56rem', '64rem'];
export const breakpoints = {
  sm: 0,
  mobile: '30em',
  tablet: '46em',
  desktop: '61.25em',
  wide: '80em',
  extrawide: '105em',
};

export default {
  space,
  fontSizes,
  breakpoints,
  maxWidths: ['48em', '60em', '80em'],
  colors: {
    theme: themeColor,
    purples,
    greens,
  },
  buttons: {
    primary: {
      color: '#fff',
      backgroundColor: themeColor,
      fontSize: fontSizes[4],
      borderColor: '#fff',
      paddingRight: space[4],
      paddingLeft: space[4],
      '&:hover': {
        backgroundColor: '#000',
      },
    },
    outline: {
      color: themeColor,
      backgroundColor: 'transparent',
      boxShadow: 'inset 0 0 2px',
    },
  },
};
