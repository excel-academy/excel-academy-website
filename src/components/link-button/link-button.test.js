import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../../theme';
import LinkButton from './link-button-css';

describe('LinkButton', () => {
  it('default renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <LinkButton to="/engineering/ux#example-list">Get Started</LinkButton>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('primary renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <LinkButton to="/engineering/ux#example-list" primary="true">Get Started</LinkButton>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('large renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <LinkButton to="/engineering/ux#example-list" large="true">Get Started</LinkButton>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('small renders correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <LinkButton to="/engineering/ux#example-list" small="true">Get Started</LinkButton>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
