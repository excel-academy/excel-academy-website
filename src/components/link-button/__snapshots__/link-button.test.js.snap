// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton default renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  overflow: hidden;
  height: 2.85em;
  padding: 0 2.5em;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #595C68;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  font-weight: 600;
  line-height: 2.85em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  white-space: nowrap;
}

.c0:active,
.c0:hover {
  background-color: #8b8e9c;
}

.c0:active {
  background-color: #2a2b31;
}

.c0:disabled {
  box-shadow: inset 0 -0.15em 0 0 rgba(0,0,0,0.15);
  opacity: 0.25;
  pointer-events: none;
}

<a
    class="c0"
    href="/engineering/ux#example-list"
  >
    Get Started
  </a>
</DocumentFragment>
`;

exports[`LinkButton large renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  overflow: hidden;
  height: 2.85em;
  padding: 0 2.5em;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #595C68;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  font-weight: 600;
  line-height: 2.85em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  white-space: nowrap;
  font-size: 1.35em;
  line-height: 2.825em;
}

.c0:active,
.c0:hover {
  background-color: #8b8e9c;
}

.c0:active {
  background-color: #2a2b31;
}

.c0:disabled {
  box-shadow: inset 0 -0.15em 0 0 rgba(0,0,0,0.15);
  opacity: 0.25;
  pointer-events: none;
}

<a
    class="c0"
    href="/engineering/ux#example-list"
    large="true"
  >
    Get Started
  </a>
</DocumentFragment>
`;

exports[`LinkButton primary renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  overflow: hidden;
  height: 2.85em;
  padding: 0 2.5em;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #595C68;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  font-weight: 600;
  line-height: 2.85em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  white-space: nowrap;
  background-color: #4bb2b6;
}

.c0:active,
.c0:hover {
  background-color: #8b8e9c;
}

.c0:active {
  background-color: #2a2b31;
}

.c0:disabled {
  box-shadow: inset 0 -0.15em 0 0 rgba(0,0,0,0.15);
  opacity: 0.25;
  pointer-events: none;
}

.c0:hover {
  background-color: #6fc2c5;
  color: #fff;
}

.c0:active {
  background-color: #2d6c6e;
}

<a
    class="c0"
    href="/engineering/ux#example-list"
    primary="true"
  >
    Get Started
  </a>
</DocumentFragment>
`;

exports[`LinkButton small renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  overflow: hidden;
  height: 2.85em;
  padding: 0 2.5em;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #595C68;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  font-weight: 600;
  line-height: 2.85em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  white-space: nowrap;
  font-size: 0.8em;
  line-height: 3em;
}

.c0:active,
.c0:hover {
  background-color: #8b8e9c;
}

.c0:active {
  background-color: #2a2b31;
}

.c0:disabled {
  box-shadow: inset 0 -0.15em 0 0 rgba(0,0,0,0.15);
  opacity: 0.25;
  pointer-events: none;
}

<a
    class="c0"
    href="/engineering/ux#example-list"
    small="true"
  >
    Get Started
  </a>
</DocumentFragment>
`;
